### PR TITLE
Refactor carry into structured module

### LINKED ISSUE #66: [Refactor] `SimulateCarry` to use `NamedTuple` for better readability
Currently, `SimulateCarry` is defined as a plain `tuple`. Passing and unpacking a plain tuple inside the `jax.lax.scan` body makes the code less readable and more fragile to structural changes. Having attribute access (e.g., `carry.time`, `carry.state`) would significantly improve code clarity.

https://github.com/hanson-hschang/Signal-System/blob/d89b9bfba89d605b3e92fe75c4910dcad58cb301/src/ss/system/_system.py#L105-L109

I propose redefining `SimulateCarry` as a `typing.NamedTuple`. JAX natively registers `NamedTuple` as a PyTree, meaning it works out-of-the-box with `lax.scan`, `jax.jit`, and auto-diff tools without any boilerplate.

```python
from typing import NamedTuple

class SimulateCarry(NamedTuple):
    time: float
    state: Shaped[Array, "batch_size ..."]
    controller_state: ControllerState | None

```

For the `body` inside the `simulate` loop, we can simply instantiate a new one at each step, which is clean and requires no special update methods:
```python
    @jax.jit
    def body(
        carry: SimulateCarry,
        random_key: PRNGKeyArray,
    ) -> tuple[SimulateCarry, SimulateStep]:
        # ... process logic ...
        next_carry = SimulateCarry(next_time, next_state, next_controller_state)
        return next_carry, (next_time, state, observation, control)
```

## Alternatives Considered & Comparison
Since JAX flattens PyTrees during compilation, all options below result in **identical XLA performance and identical gradient tracking** for differentiable filtering.

### 1. `equinox.Module`
Fits well if standardizing on Equinox, but requires importing `dataclasses.replace` for updates and makes `equinox` a strict dependency just for a data container.

```python
import equinox as eqx
from dataclasses import replace

class SimulateCarry(eqx.Module):
    time: float
    state: Array
    controller_state: ControllerState | None

# Update inside scan:
next_carry = replace(carry, time=next_time, state=next_state, controller_state=next_controller_state)

```

### 2. `flax.struct.dataclass`
Provides a very clean built-in `.replace()` method, but requires adding `flax` as a dependency, which is unnecessary overhead if the project otherwise uses Equinox or pure JAX.

```python
from flax import struct

@struct.dataclass
class SimulateCarry:
    time: float
    state: Array
    controller_state: ControllerState | None

# Update inside scan:
next_carry = carry.replace(time=next_time, state=next_state, controller_state=next_controller_state)

```

## Conclusion
Given that we update all three fields (`time`, `state`, `controller_state`) at every step of the simulation, the lightweight, native, and dependency-free nature of `NamedTuple` makes it the best fit.

### PR DESCRIPTION
**Pull request by skim0119**
closes #66 

## Summary

- replace the tuple-based `SimulateCarry` with an `eqx.Module`
- use named carry fields throughout the simulation scan
- replace MPPI's tuple-based `RolloutCarry` with an `eqx.Module`

This removes positional carry unpacking.

## Testing

- [x] test
- [x] formatting
- [x] typing

### CODE REVIEW

**Comment by hanson-hschang** [comment]
@skim0119 

You work on it so fast... Could you also do `FilteringCarry` then XD. Im not sure if the following should be combined together as one pull request. I want to make sure the naming of the `simulation` and `filtering` is consistent. This is the current status:

| `simulate` | `filtering` | comments |
| ------------ | ---------- | :---------- |
| `body`       | `step` | I think `step` is better |
| `SimulateCarry` | `FilteringCarry` | `FilteringCarry` is still an type alias for now |
| `SimulateStep` | `Float[Array, "batch_size state_dim"]` | the filtering side should be a type alias as well |

Also, the return of `step` in the filtering should also include `next_time` rather than just `posterior`, right? It should be like
```python
return next_carry, (next_time, posterior)
```

**Comment by skim0119** [comment]
@hanson-hschang added changes

**Comment by skim0119** [comment]
All tests added in this PR are now wrapped in test classes for consistency.

**Comment by hanson-hschang** [comment]
LGTM
